### PR TITLE
fix: Fix memory leak of ChatView.Coordinator and ChatWebView

### PR DESCRIPTION
**Pull request by nesium**
When you switched back and forth between e.g. "Direct Messages" and any chat in the sidebar for each selection of a chat a   ChatView.Coordinator and WKWebView was leaked.

### CODE REVIEW

File: Prose/ProseLib/Sources/ConversationFeature/Chat/Chat.swift [hunk 1]
```diff
@@ -199,11 +199,14 @@ struct ChatView: NSViewRepresentable {
...
       .drop(while: { $0 == nil })
       .removeDuplicates()
-      .sink { menuState in
+      .sink { [weak coordinator = context.coordinator] menuState in
+        guard let coordinator = coordinator else {
+          return
+        }
         if let menuState: MessageMenuState = menuState {
-          self.showMenu(menuState, on: webView, context: context)
+          self.showMenu(menuState, on: webView, coordinator: coordinator)
```
Comment (RemiBardon) [inline Prose/ProseLib/Sources/ConversationFeature/Chat/Chat.swift:207]: Isn't it a problem that `webView` is sent here? Shouldn't we add `weak webView` too?

Comment (nesium) [reply Prose/ProseLib/Sources/ConversationFeature/Chat/Chat.swift:207]: No ;)

Comment (RemiBardon) [reply Prose/ProseLib/Sources/ConversationFeature/Chat/Chat.swift:207]: The answer is no: The coordinator was captured, which prevented the `AnyCancellable`s from being canceled. Now they are, which frees the web view.

**Review by RemiBardon — APPROVED** [review]
Thank you for this catch!

**Pull request merged**